src/LetsTrace/FieldExtensions.cs

```diff
@@ -51,33 +51,38 @@ public static Field ToField(this KeyValuePair<string, object> kv)
             if (kv.Value is byte[]) {
                 return kv.Value.ToField<byte[]>(kv.Key);
             }
-
-            var typeCode = Type.GetTypeCode(kv.Value.GetType());
-            
-            switch(typeCode)
+            if (kv.Value == null) {
+                return new Field<string> { Key = kv.Key, Value = "" };
+            }
+            else
```

Comment:
I would remove the else branch since the if-case already has a return in it. Reduces nesting by one level. And makes it similar to the byte-array case above.